src/components/SpotifyPlayer.tsx

```
@@ -39,78 +43,37 @@ function SpotifyWaveform({
-        <div
-          key={i}
-          className="w-0.5 bg-white/80 rounded-full transition-all duration-200 ease-out"
-          style={{
-            height: `${heights[i] || 20}%`,
-          }}
-        />
-      ))}
+    <div className={`flex items-end justify-evenly w-full ${compact ? 'h-6' : 'h-12'}`}>
+      {bars.map((_, i) => {
+        const position = i / barCount;
+        const height = useTransform(time, (t) => {
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)